operator/controllers/operator/eventexporter_controller.go

```diff
@@ -0,0 +1,210 @@
+			Name:      eventExporter.Name,
+			Namespace: eventExporter.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: eventExporter.APIVersion,
+					Kind:       eventExporter.Kind,
+					Name:       eventExporter.Name,
+					UID:        eventExporter.UID,
+				},
+			},
+			Labels: map[string]string{
+				"version":  eventExporter.Spec.Version,
+				"md5-data": newMd5,
+			},
+		},
+		Data: map[string]string{"config.yaml": eventExporter.Spec.Data},
+	}
+
+	if err := r.Client.Create(ctx, &configmap); err != nil {
+		log.Error(err, "failed to create eventexporter's configmap")
+		return true, err
+	}
+
+	return true, nil
+}
+
+func (r *EventExporterReconciler) checkState(ctx context.Context, log logr.Logger, eventExporter *operatorv1alpha1.EventExporter) error {
+	overlay := operatorv1alpha1.EventExporterStatus{}
+	deployment := apps.Deployment{}
+	errCol := new(kubernetes.ErrorCollector)
+
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: eventExporter.Namespace, Name: eventExporter.Name + "-eventexporter"}, &deployment); err != nil && !apierrors.IsNotFound(err) {
+		errCol.Collect(fmt.Errorf("failed to get deployment: %w", err))
+	} else {
+		overlay.Conditions = deployment.Status.Conditions
+		overlay.AvailableReplicas = deployment.Status.AvailableReplicas
+	}
+
+	if apiequal.Semantic.DeepDerivative(overlay, eventExporter.Status) {
+		log.Info("Status keeps the same as before")
+		return errCol.Error()
+	}
+
+	if err := r.updateStatus(ctx, eventExporter, overlay, errCol); err != nil {
+		errCol.Collect(fmt.Errorf("failed to update status of EventExporter: %w", err))
+	}
+
+	log.Info("updated Status sub resource")
+
+	return errCol.Error()
+}
+
+func (r *EventExporterReconciler) updateStatus(ctx context.Context, eventExporter *operatorv1alpha1.EventExporter,
+	overlay operatorv1alpha1.EventExporterStatus, errCol *kubernetes.ErrorCollector) error {
+	// avoid resource conflict
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		if err := r.Client.Get(ctx, client.ObjectKey{Name: eventExporter.Name, Namespace: eventExporter.Namespace}, eventExporter); err != nil {
+			errCol.Collect(fmt.Errorf("failed to get EventExporter: %w", err))
+		}
+
+		eventExporter.Status = overlay
+		eventExporter.Kind = "EventExporter"
+		if err := kubernetes.ApplyOverlay(eventExporter, &operatorv1alpha1.EventExporter{Status: overlay}); err != nil {
+			errCol.Collect(fmt.Errorf("failed to apply overlay: %w", err))
+		}
+
+		if err := r.Status().Update(ctx, eventExporter); err != nil {
+			errCol.Collect(fmt.Errorf("failed to update status of EventExporter: %w", err))
+		}
+		return errCol.Error()
+	})
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *EventExporterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&operatorv1alpha1.EventExporter{}).
+		Owns(&apps.Deployment{}).
+		Owns(&core.ConfigMap{}).
```

Comment:
I think if we do not add predicates, all deploy/configmaps change will trigger  eventExporter controller to reconcile. So we can remove them.

Reply:
Thx, updated!